distros/noetic/overrides.nix

```diff
@@ -175,4 +175,56 @@ rosSelf: rosSuper: with rosSelf.lib; {
       wrapQtApp "$out/lib/rqt_topic/rqt_topic"
     '';
   });
+
+  costmap-converter = rosSuper.costmap-converter.overrideAttrs ({
+    patches ? [], ...
+  }: {
+    # fix: invalid new-expression of abstract class type ‘BlobDetector’
+    # https://github.com/rst-tu-dortmund/costmap_converter/pull/40
+    patches = patches ++ [ (self.fetchpatch {
+      url = "https://github.com/rst-tu-dortmund/costmap_converter/pull/40.patch";
+      hash = "sha256-xIjxFsyWbHzIhuthGivpZAA2xdFQm7g68bsNjcS8cYM=";
+    }) ];
+  });
+
+  libg2o = rosSuper.libg2o.overrideAttrs ({
+    propagatedBuildInputs ? [],...
+  }: {
+    propagatedBuildInputs = propagatedBuildInputs ++ [ self.openblas ];
+  });
+
+  teb-local-planner = rosSuper.teb-local-planner.overrideAttrs ({
+    patches ? [], cmakeFlags ? [], buildInputs ? [],...
+  }: {
+    # fix: boost sign
+    # https://github.com/rst-tu-dortmund/teb_local_planner/pull/413
+    patches = patches ++ [ (self.fetchpatch {
+      url = "https://github.com/rst-tu-dortmund/teb_local_planner/pull/413.patch";
+      hash = "sha256-td4yejakcXUH6b6wUa2d85R4dyegc5pbZLDEzTTz/x8=";
+    }) ];
+    cmakeFlags = cmakeFlags ++ [ "-DCMAKE_INCLUDE_PATH=${self.suitesparse}/lib" ];
```

Comment:
This is necessary because `FindSUITESPARSE.cmake` uses `find_path()` to search for the directory containing `libcholmod.so`, but this is primarily designed to find include directories and doesn't search `${CMAKE_PREFIX_PATH}/lib`. The correct way is to use `find_library()`, but this returns the file path rather than the directory, so would require some restructuring of the script.

If we don't want to bother with that, it would be better to pass `-DSUITESPARSE_LIBRARY_DIR=${self.suitesparse}/lib`.